builtin/credential/jwt/path_login.go

```diff
@@ -194,11 +205,99 @@ func (b *jwtAuthBackend) pathLogin(ctx context.Context, req *logical.Request, d
 		return nil, err
 	}
 
+	if err := b.applyCelRoles(ctx, req, allClaims, auth); err != nil {
```

Comment:
Ah interesting, I think this is different than I'd expect based on discussion in https://github.com/openbao/openbao/issues/493#issuecomment-2549033976 -- though, reading the RFC more carefully, I think there's a mismatch between RFC sections.

In "User-facing Description":

> **Administrators** can define CEL extensions under a new endpoint, `auth/{oidc|jwt}/cel/roles/:name`, to be executed during authentication. CEL roles evaluate incoming claims and generate `logical.Auth{}` instances having appropriate token policies.

Namely, the "executed during authentication" makes me think it supplements the existing role-based system, as is the case in this code. However, I thought we had decided on _replacing_ the role system entirely, which seems to match what is in the "Technical Description" section:

> CEL roles integrate with Vault's Auth engine (OIDC and JWT) to dynamically evaluate claims. Instead of using static role-to-policy assignment with `bound_claim`, CEL programs validate claims and return a `logical.Auth` instance containing dynamically assigned token policies.

(from the "Instead of using static role-to-policy..." portion of that quote).

My 2c., but I think I'd prefer the latter to the former as it is simpler to reason about and has fewer moving parts.  

---

In particular, though not articulated in the RFC, I think I'd have expected a CEL/regular role _name_ to be globally unique (across both types). Login (with a role name) would then lookup the single unique role (whether regular or CEL) and evaluate just that one role on the request. 

Or, perhaps easier, rather than globally unique, have a `cel/login/:role` path, which logs in with the specified CEL role.